src/nodes/treeensembleclassifier.cc

```diff
@@ -0,0 +1,236 @@
+		int64_t nodeId = node_node_ids[i];
+		trees[treeId][nodeId]->child_true = trees[treeId][node_true_ids[i]];
+		trees[treeId][nodeId]->child_false = trees[treeId][node_false_ids[i]];
+		trees[treeId][nodeId]->feature = node_feature_ids[i];
+		trees[treeId][nodeId]->threshold = node_thresholds[i];
+		trees[treeId][nodeId]->mode = node_modes[i];
+	}
+
+	return trees;
+}
+
+/* Body of the node implementing function */
+void TreeEnsembleClassifier::print(std::ostream &dst) const
+{
+	std::unordered_map<int64_t, Tree> trees = generateTreeMap();
+
+	for( int64_t class_id : class_ids )
+	{
+		INDT_1 << "float result_class_" << class_id << " = 0.0f;" << std::endl;
+	}
+
+	for( auto& [tree_id, tree] : trees )
+	{
+		INDT_1 << "/* Tree " << tree_id << " */" << std::endl;
+		std::stack<std::shared_ptr<TreeNode>> node_stack;
+		node_stack.push(tree.at(0));
+		std::shared_ptr<TreeNode> current_node = tree.at(0);
+
+		do
+		{
+			if( !current_node )
+			{
+				current_node = node_stack.top();
+				node_stack.pop();
+			}
+
+			if( current_node->is_leaf )
+			{
+				for( size_t i = 0; i < current_node->class_ids.size(); ++i)
+				{
+					INDT(node_stack.size()) << "result_class_" << current_node->class_ids[i] << " += " << current_node->weights[i] << ";" << std::endl;
+				}
+				current_node.reset();
+			}
+			else
+			{
+				node_stack.push(current_node);
+				std::shared_ptr<TreeNode> nextNode;
+				if (current_node->child_true)
+				{
+					// features[0] because the batch size is always 1
+					INDT(node_stack.size() - 1) << "if( features[0][" << current_node->feature << "] " << convertMode(current_node->mode) << " " << current_node->threshold << " ) {" << std::endl;
+					nextNode = current_node->child_true;
+					current_node->child_true.reset();
+				}
+				else if (current_node->child_false)
+				{
+					INDT(node_stack.size() - 1) << "} else {" << std::endl;
+					nextNode = current_node->child_false;
+					current_node->child_false.reset();
+				}
+				else
+				{
+					if (node_stack.size() > 1)
+					{
+						INDT(node_stack.size() - 1) << "}" << std::endl;
+					}
+					nextNode.reset();
+					node_stack.pop();
+				}
+				current_node = nextNode;
+			}
+		}
+		while( !node_stack.empty() );
+	}
+
+	for( int64_t class_id : class_ids )
+	{
+		if( unused_class_id == class_id )
+		{
+			continue;
+		}
+		INDT_1 << "probabilities[" << class_id << "] = ";
+		
+		if(post_transform == "NONE")
+		{
+			dst << "result_class_" << class_id << ";" << std::endl;
+		}
+		else if(post_transform == "LOGISTIC")
+		{
+			dst << "1.0f / (1.0f + exp(result_class_" << class_id << "));" << std::endl;
+		}
+	}
+
+	if( unused_class_id != -1 )
+	{
+		INDT_1 << "probabilities[" << unused_class_id << "] = 1.0f - (";
+		for( size_t i = 0; i < class_ids.size(); ++i )
+		{
+			if( class_ids[i] == unused_class_id )
+			{
+				continue;
+			}
+			dst << "probabilities[" << class_ids[i] << std::string("]") + (i < class_ids.size() - 1 && unused_class_id != i + 1 ? " + " : "");
```

Comment:
One more sign mismatch
```suggestion
			dst << "probabilities[" << class_ids[i] << std::string("]") + (i < class_ids.size() - 1 && unused_class_id != (int)i + 1 ? " + " : "");
```

Reply:
Sorry about this. Now _I_ don't understand how I didn't catch this yesterday :/

Reply:
I introduced that one by fixing an issue from yesterday